controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -671,6 +672,20 @@ private long findSegmentSplitsMerges(List<StreamSegmentRecord> referenceSegments
                       });
     }
 
+    @Override
+    public CompletableFuture<Boolean> isStreamCutValidForTruncation(final Map<Long, Long> streamCut, Map<Long, Long> previousStreamCut) {
```

Comment:
we dont necessarily need a public method here. this can be done as part of precondition check for updating a subscriber streamcut internally. that will allow for get and update post validation. 
otherwise you are validating. and then refetchng subscriber's record and updating it which may mean the condition may have changed. 
Also the implementation of this method can be optimized slightly. 

PS: I have made those changes in my PR.

Reply:
i am making the change in my PR to make the optimization to this method and its invocation in streamMetadataTasks.

Reply:
Ok

Reply:
changed this to a private method